src/lib/operators/get_table.cpp

```diff
@@ -239,8 +239,38 @@ std::shared_ptr<const Table> GetTable::_on_execute() {
     ++output_chunks_iter;
   }
 
+  // Check if a table index is irrelevant because of pruning.
+  const auto indexed_column_or_chunks_pruned = [this](const std::shared_ptr<AbstractTableIndex>& table_index) {
+    // Check if indexed ColumnID has been pruned.
+    const auto indexed_column_id = table_index->get_indexed_column_id();
+    if (std::find(_pruned_column_ids.begin(), _pruned_column_ids.end(), indexed_column_id) !=
+      _pruned_column_ids.end()) {
+      return true;
+    }
+
+    const auto indexed_chunk_ids = table_index->get_indexed_chunk_ids();
+
+    // Early out if index is empty.
+    if (indexed_chunk_ids.empty()) {
+      return false;
+    }
+
+    auto sorted_indexed_chunk_ids = std::vector<ChunkID>(indexed_chunk_ids.begin(), indexed_chunk_ids.end());
+    std::sort(sorted_indexed_chunk_ids.begin(), sorted_indexed_chunk_ids.end());
+    DebugAssert(std::is_sorted(_pruned_chunk_ids.begin(), _pruned_chunk_ids.end()),
+                "Pruned ChunkIDs should be sorted.");
+
+    // Check if the indexed chunks have been pruned.
+    return (std::includes(_pruned_chunk_ids.begin(), _pruned_chunk_ids.end(),
+                          sorted_indexed_chunk_ids.begin(), sorted_indexed_chunk_ids.end()));
+  };
+
```

Comment:
Guess that's a matter of taste, but you can also build a set
```c++
const auto pruned_chunk_ids = std::unordered_set{_pruned_chunk_ids.cbegin(), _pruned_chunk_ids.cend()};
```
and check with if all indexed chunks are contained in this set:
```c++
return std::all_of(indexed_chunk_ids.cbegin(), indexed_chunk_ids.cend(),
                   [&](const auto chunk_id) { return pruned_chunk_ids.contains(chunk_id); });
```

Saves you one LoC for sorting ;)

Reply:
Anyways, you can use `.cbegin()`/`.cend()` more often and the parantheses around `std::includes` are superfluous.